lib/tasks/expiration.rake

```diff
@@ -0,0 +1,21 @@
+require 'iso_8601_duration'
+
+namespace :expiration do
+  desc "Watch for expired variables and rotate"
+  task :watch => :environment do
+
+    #TODO: perhaps use clockwork / similar rather than a sleep loop
+    while true
+      Secret.freshly_expired.each do |secret|
+        puts "Secret #{secret[:resource_id]} expired!  Resetting it..."
```

Comment:
When this expired secret is found and used to create a delayed job (or similar), an update should be made to the secret to prevent another delayed job from being created in the next loop cycle.

Reply:
so actually, on reflection, one counterpoint to using delayedjob idea, is that the current implementation will already handle retries.  if a variable does not get rotated, it will get picked up again as needing rotation on the next loop iteration (after 1s).  as long we handle errors properly so that the loop process itself doesn't die, that might be enough?  are there any additional features we need out of delayedjob.  it does require a DB back so the complexity cost isn't trivial....

Reply:
Maybe `conjurctl` should be responsible for terminating the parent process if any of the sub-processes dies. Because we can't let any of them stop running (the main server process, authn-local or rotation)

Reply:
Also, see the above comment about ensuring that the database is writeable before this loop is allowed to run. That will ensure that standbys / followers don't try and run the rotator loop, which could cause multiple processes to be rotating the same backend password at the same time (which would be bad).

Reply:
> Maybe conjurctl should be responsible for terminating the parent process if any of the sub-processes dies. Because we can't let any of them stop running (the main server process, authn-local or rotation)

Can you elaboarate on this one?  How were you thinking the parent process would monitor the sub-process?

Reply:
Well, instead of `Process.waitall` (https://github.com/cyberark/conjur/blob/master/bin/conjur-cli.rb#L81), when any process exits start terminating the other ones as well.

This feels like a separate story.

Reply:
It looks like @dividedmind is making a similar suggestion with something called "rufus scheduler"